Library/Homebrew/exceptions.rb

```diff
@@ -524,9 +526,16 @@ def initialize(cause)
 
 # raised by safe_system in utils.rb
 class ErrorDuringExecution < RuntimeError
-  def initialize(cmd, args = [])
-    args = args.map { |a| a.to_s.gsub " ", "\\ " }.join(" ")
-    super "Failure while executing: #{cmd} #{args}"
+  def initialize(cmd, status:, output: nil)
+    s = "Failure while executing; `#{cmd.shelljoin.gsub(/\\=/, "=")}` exited with #{status.exitstatus}."
+
+    unless [*output].empty?
+      s << " Here's the output:\n"
+      s << output.map { |(type, line)| (type == :stderr) ? Formatter.error(line) : line }.join
```

Comment:
Can this go into a multiline `map` with an `if`/`else`? It's quite hard to parse as-is. Thanks!